src/main/java/com/muzika/homeworksb/model/TaskHistory.java

```diff
@@ -0,0 +1,67 @@
+package com.muzika.homeworksb.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.time.LocalDateTime;
+
+/**
+ * Tracks changes made to Todo tasks.
+ */
+@Data
+@Entity
+@Table(name = "task_history")
+public class TaskHistory {
+
+    /**
+     * Unique identifier for the TaskHistory entry
+     */
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    /**
+     * Reference to related Todo item
+     */
+    @EqualsAndHashCode.Exclude
+    @ToString.Exclude
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "todo_id", nullable = false)
+    @JsonIgnore
+    private Todo todo;
+
+    /**
+     * to simplify the task you can use Todo toString() method response and store here. For advanced solution - convert Todo entity to JSON format and store it
+     */
+    @Column(name = "old_state")
+    private String oldState;
+
+    /**
+     * String (same as for oldState)
+     */
+    @Column(name = "new_state")
+    private String newState;
+
+    /**
+     * The timestamp when the change was made
+     */
+    @Column(name = "changed_date")
+    private LocalDateTime changeDate;
```

Comment:
JFYI, in some cases you may use [Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html) instead of LocalDateTime. But in our case any option is valid